python_transport/wirepas_gateway/transport_service.py

```diff
@@ -233,6 +247,7 @@ def on_sink_disconnected(self, name):
         self.logger.info("Sink disconnected, sending new configs")
         self._send_asynchronous_get_configs_response()
 
+    @deferred_thread
```

Comment:
Could we run into a race condition when calling a few resources used by these threads? For example, the sink_manager.get_sink?

Reply:
No it shouldn't. The only shared ressource being the dbus object. This is the only place where the different Threads will release the GIL (IO blocking). So Threads will get the CPU only when one is waiting on dbus and this part should be thread safe.
I did a test by creating long delay on sink service to serve the request and doing multiple time the same request on MQTT. Threads are all blocked on dbus and get access to it one by one.
Of course their is still a possibility of sink service disapearing but it is exactly the same with current implementation and already handled.

Reply:
good with me!